### DISCUSSION TITLE
How is this project importing typescript files with file extension?

### DISCUSSION THREAD

**Discussion by JumpLink** [eemeli/yaml#342] [General]
I want to publish a module for browser, Node.js and Deno as it is the case with this module.

However, Deno wants me to import the files with the file extension as it is also done here.

Example on Node:

```ts
import { myModule } from './my-module';
```

Example on Deno:

```ts
import { myModule } from './my-module.ts';
```

There is also an issue for this on typescript and there seems to be no solution there yet: https://github.com/microsoft/TypeScript/issues/37582

Hence the question: How was this solved in yaml?

**Comment by JumpLink** [comment]
Oh sorry, i didn't see that here is imported with `.js`. That seems to be the only way?

**Reply by eemeli** [reply]
That's the way that is supported by both TypeScript and Node.js. At least previously the loader used by Deno did not support loading `.ts` files by using `.js` as the extension. I would like to make this library more compatible with Deno, but have not found a low-impact way of doing so.

**Comment by Logan0112** [comment]
Yo quiero
